telemetry/data_cluster_hardware.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+import (
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/sqlexec"
+)
+
+var (
+	// Explicitly list all allowed field names to avoid potential leaking sensitive info when new fields are added later.
+	sortedCPUAllowedFieldNames    = []string{"cpu-logical-cores", "cpu-physical-cores", "cpu-frequency", "cache", "cpu-vendor-id", "l1-cache-size", "l1-cache-line-size", "l2-cache-size", "l2-cache-line-size", "l3-cache-size", "l3-cache-line-size"}
+	sortedDiskAllowedFieldNames   = []string{"fstype", "opts", "total", "free", "used", "free-percent", "used-percent"} // path is not included
+	regexDiskAllowedNames         = regexp.MustCompile(`^(disk\d+s\d+|rootfs|devtmpfs|sd[a-z]\d*|vd[a-z]\d*|hd[a-z]\d*|nvme\d+(n\d+(p\d)?)?)$`)
+	sortedDiskAllowedPaths        = []string{"/", "/boot", "/dev", "/private/var/vm", "/System/Volumes/Data"}
+	sortedMemoryAllowedFieldNames = []string{"capacity"}
+)
+
+func init() {
+	sort.Strings(sortedCPUAllowedFieldNames)
+	sort.Strings(sortedDiskAllowedFieldNames)
+	sort.Strings(sortedDiskAllowedPaths)
+	sort.Strings(sortedMemoryAllowedFieldNames)
+}
+
+type clusterHardwareItem struct {
+	InstanceType   string                       `json:"instanceType"`
+	ListenHostHash string                       `json:"listenHostHash"`
+	ListenPort     string                       `json:"listenPort"`
+	CPU            map[string]string            `json:"cpu,omitempty"`
+	Memory         map[string]string            `json:"memory,omitempty"`
+	Disk           map[string]map[string]string `json:"disk,omitempty"`
+}
+
+func normalizeDiskName(name string) string {
+	const prefix = "/dev/"
```

Comment:
Could we remove `const ` or move this line outside this function(move it below line35)?

Reply:
Maybe it is better to be placed here to keep contexts close, otherwise a new comment maybe needed to introduce it.